src/main/java/org/jetbrains/research/intellijdeodorant/JDeodorantFacade.java

```diff
@@ -1,24 +1,86 @@
 package org.jetbrains.research.intellijdeodorant;
 
+import com.intellij.ide.errorTreeView.impl.ErrorTreeViewConfiguration;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.compiler.CompileContext;
+import com.intellij.openapi.compiler.CompileScope;
+import com.intellij.openapi.compiler.CompileStatusNotification;
+import com.intellij.openapi.compiler.CompilerManager;
 import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiClass;
 import com.intellij.psi.PsiVariable;
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.research.intellijdeodorant.core.ast.ASTReader;
 import org.jetbrains.research.intellijdeodorant.core.ast.ClassObject;
 import org.jetbrains.research.intellijdeodorant.core.ast.MethodObject;
 import org.jetbrains.research.intellijdeodorant.core.ast.SystemObject;
 import org.jetbrains.research.intellijdeodorant.core.ast.decomposition.cfg.*;
 import org.jetbrains.research.intellijdeodorant.core.distance.*;
-import org.jetbrains.research.intellijdeodorant.core.distance.DistanceMatrix;
-import org.jetbrains.research.intellijdeodorant.core.distance.MoveMethodCandidateRefactoring;
-import org.jetbrains.research.intellijdeodorant.core.distance.MySystem;
-import org.jetbrains.research.intellijdeodorant.core.distance.ProjectInfo;
 import org.jetbrains.research.intellijdeodorant.ide.refactoring.typeStateChecking.TypeCheckEliminationGroup;
+import org.jetbrains.research.intellijdeodorant.ide.ui.AbstractRefactoringPanel;
 
 import java.util.*;
 
 public class JDeodorantFacade {
+    public static void runAfterCompilationCheck(ProjectInfo projectInfo, Task task) {
+        ApplicationManager.getApplication().invokeLater(() -> {
+            List<PsiClass> classes = projectInfo.getClasses();
```

Comment:
When https://github.com/JetBrains-Research/IntelliJDeodorant/pull/35 is ready, we should get here classes corresponding to the user configuration (all classes / all classes opened in editor / current class opened in the editor)

Reply:
Also, in order to achieve that, class `SettingsPanel` should get two methods such as `getAllObjectsOpenedInEditor()` -- one that returns `Set<PsiClass>` (in order to use it here), and another - that returns `Set<ClassObject>` (apply `ASTReader.processTypeDeclaration` to the result of the first method).